File: Easy/242-Valid-Anagram.cpp
// https://leetcode.com/problems/valid-anagram/submissions/960991852/
// Beats 46.1% runtime and 48.5% memory

class Solution {
#include <bits/stdc++.h>
public:
    bool isAnagram(string s, string t) {
        using namespace std;
        if (s.size() != t.size()){
            return false;
        }
        sort(s.begin(), s.end());
        sort(t.begin(), t.end());
        return s == t;
    }
};

File: Easy/242-Valid-Anagram.rs
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() {
            return false;
        }

        let mut s_chars: Vec<char> = s.chars().collect();
        let mut t_chars: Vec<char> = t.chars().collect();
        
        s_chars.sort();
        t_chars.sort();

        s_chars == t_chars
    }
}
